### PR TITLE
Add support for divideSeriesLists

### PR DESCRIPTION
**Pull request by shanson7**

### CODE REVIEW

File: docs/graphite.md [hunk 1]
```diff
@@ -35,7 +35,8 @@ aliasSub(seriesList, pattern, replacement) seriesList |              | Stable
 averageSeries(seriesLists) series                     | avg          | Stable
 consolidateBy(seriesList, func) seriesList            |              | Stable
 diffSeries(seriesLists) series                        |              | Stable
-divideSeries(seriesList, dividend, divisor) seriesList|              | Stable
+divideSeries(dividend, divisor) seriesList            |              | Stable
```
Comment (Spellchaser) [inline docs/graphite.md]: Aha, this was glossed over from before

File: expr/func_divideserieslists.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+		divisor := divisors[i]
+		out := pointSlicePool.Get().([]schema.Point)
+		for i := 0; i < len(dividend.Datapoints); i++ {
+			p := schema.Point{
+				Ts: dividend.Datapoints[i].Ts,
+			}
+			if divisor.Datapoints[i].Val == 0 {
+				p.Val = math.NaN()
+			} else {
+				p.Val = dividend.Datapoints[i].Val / divisor.Datapoints[i].Val
```
Comment (Spellchaser) [inline expr/func_divideserieslists.go]: This doesn't seem equivalent to safeDiv. What about dividing by 0

File: expr/func_divideserieslists_test.go [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+				Tags: map[string]string{
+					"name": "DivideSeriesLists(foo-2;a=2;b=2;b=2,overbar-2;a=3;b=2;c=1)",
+				},
+			},
+		},
+		t,
+	)
+}
+
+func testDivideSeriesLists(name string, dividend, divisor []models.Series, out []models.Series, t *testing.T) {
```
Comment (Spellchaser) [inline expr/func_divideserieslists_test.go]: No test cases cover division by 0

File: expr/func_divideserieslists.go (hunk 2 above)
Comment (shanson7) [reply expr/func_divideserieslists.go]: golang handles that : https://play.golang.org/p/lpkONXNAkcq

As far as I can google, this isn't well-defined (this implementation was inspired from `divideSeries`).

Comment (Spellchaser) [reply expr/func_divideserieslists.go]: Ahh I was trying with ints :) forgot it's all floats
NaN is diff from Inf though so it'll be slightly different

**Comment by Spellchaser** [comment]
Adding test for division by 0 to explicitly state it'll return golang's Inf not NaN like graphite-web would be a form of documentation.

**Pull request merged**